src/MongoDB.Driver.Core/Core/Authentication/GssapiAuthenticator.cs

```diff
@@ -189,6 +190,7 @@ public GssapiMechanism(string serviceName, bool canonicalizeHostName, string rea
                 _realm = realm;
                 _username = Ensure.IsNotNullOrEmpty(username, nameof(username));
                 _password = password;
+                _dnsResolver = new DnsClientWrapper();
```

Comment:
Should we use a `DnsClientWrapper` singletion?

Replace `new DnsClientWrapper()` with `DnsClientWrapper.Instance`?

Reply:
I asked about it here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/84/files#r351291980
Singleton sounds good to me. Will add it in the next commit

Reply:
I saw your comment. That's why I asked, because you made the comment but the code still wasn't using a singleton.